lib/cli.ex

```diff
@@ -22,24 +22,24 @@ defmodule Onigumo.CLI do
 
             case switches do
               [] -> module.main(working_dir)
-              _ -> usage_message()
+              _ -> usage_message("invalid OPTIONS")
```

Comment:
```suggestion
              _ -> usage_message("incompatible OPTIONS")
```

in case you don’t like „invalid”.

---

It looks like there is no way to get the exact switch literal used in _argv_. We can’t discern whether the user provided `-h` or `--help`.

It may, however, not be a problem. The help message always lists a long version along with its short alias. Each switch has the long version.

I mean, printing “invalid option help” or “invalid option --help” may not be that bad even if the user provided `-h`. It may look slightly confusing, but at the same time it provides useful additional information.

There is a function `OptionParser.to_argv/1` that translates argument names and values to their CLI counterparts. It can take the `invalid` keyword list returned by `OptionParser.parse/3`.

```elixir
iex(1)> OptionParser.to_argv([help: true])
["--help"]
```

Reply:
thank you I think `OptionParser.to_argv` will be good choice to both `help` and `invalid`

Reply:
Ok it seems it cannot be used for `invalid`